src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestOptions.cs

```diff
@@ -80,5 +83,33 @@ public void Set<TValue>(HttpRequestOptionsKey<TValue> key, TValue value)
         {
             Options[key.Key] = value;
         }
+
+        private string DebuggerToString()
+        {
+            return $"Count = {Options.Count}";
+        }
+
+        private sealed class HttpRequestOptionsDebugView(HttpRequestOptions options)
+        {
+            private readonly Dictionary<string, object?> _collection = options.Options;
+
+            [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+            public KeyValuePair<string, string?>[] Items
+            {
+                get
+                {
+                    var kvpArr = new KeyValuePair<string, string?>[_collection.Count];
+                    int i = 0;
+
+                    foreach (var kvp in _collection)
+                    {
+                        kvpArr[i] = new KeyValuePair<string, string?>(kvp.Key, kvp.Value?.ToString());
+                        i++;
+                    }
+
+                    return kvpArr;
+                }
```

Comment:
This shouldn't `ToString` the values; it should return them to the debugger as-is.
```C#
private sealed class HttpRequestOptionsDebugView(HttpRequestOptions options)
{
    [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
    public KeyValuePair<string, object?>[] Items
    {
        get
        {
            var items = new KeyValuePair<string, object?>[options.Count];
            options.CopyTo(items, 0);
            return items;
        }
    }
}
```